src/API/Base.php

```diff
@@ -460,14 +460,79 @@ public static function update_merchant_feed( $merchant_id, $feed_id, $args ) {
 
 
 	/**
-	 * Get a specific merchant's feed using the given arguments.
+	 * Get a specific merchant feed using the given arguments.
 	 *
 	 * @param string $merchant_id The merchant ID the feed belongs to.
 	 * @param string $feed_id     The ID of the feed.
 	 *
-	 * @return mixed
+	 * @return mixed|null
+	 *
+	 * @throws \Exception PHP Exception.
 	 */
 	public static function get_merchant_feed( $merchant_id, $feed_id ) {
+		try {
+
+			$feeds = self::get_merchant_feeds( $merchant_id );
+
+			if ( 'success' !== $feeds['status'] ) {
+				return null;
```

Comment:
I think its best to throw an exception here as well (and expect that exception where this is called). That way you also have 1 single return type from the method